crates/tests/fixtures/files/arrays.fe

```diff
@@ -0,0 +1,53 @@
+use std::evm
+
+contract Foo {
+    my_array: Array<u256, 10>
+
+    pub fn __init__(mut self) {
+        self.my_array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+    }
+
+    pub fn get_from_storage(self, index: u256) -> u256 {
+        return self.my_array[index]
+    }
+
+    pub fn get_from_memory(index: u256) -> u256 {
+        let my_array: Array<u256, 10> = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        return my_array[index]
+    }
+}
+
+#test
+unsafe fn test_foo() {
+    let mut ctx: Context = Context()
+    let foo: Foo = Foo.create(ctx, 0)
+    assert foo.get_from_storage(index: 3) == 3
+    assert Foo::get_from_memory(index: 3) == 4
+}
+
+contract ArrayIndexOob {
+    pub unsafe fn __call__() {
+        let mut my_array: Array<u256, 4> = [0, 1, 2, 3]
+        let my_num: u256 = my_array[4]
+
+        // array index oob revert only tirggered if the value is used due to optimization
+        evm::sstore(offset: 0, value: my_num)
+    }
+}
+
+#test
+unsafe fn test_array_oob() {
+    let mut ctx: Context = Context()
+    let array_index_oob: ArrayIndexOob = ArrayIndexOob.create(ctx, value: 0)
+
+    // verify that the call reverts
+    assert evm::call(
+        gas: 1000, 
+        addr: address(array_index_oob), 
+        value: 0, 
+        input_offset: 0, 
+        input_len: 0, 
+        output_offset: 0, 
+        output_len: 0
+    ) == 0
```

Comment:
It looks like the old tests also covered that the correct error code is returned whereas these tests simply check that it reverts but do not check the actual error that was returned. Maybe we can add this back?